pkg/artifactory/resource_artifactory_remote_repository.go

```diff
@@ -531,6 +541,10 @@ func resourceRemoteRepositoryExists(d *schema.ResourceData, m interface{}) (bool
 	key := d.Id()
 	_, resp, err := c.V1.Repositories.GetRemote(context.Background(), key)
 
+	if resp != nil {
```

Comment:
The equality check should be `resp == nil` but check `err` first. If you check error first you probably never need to check if `resp` is `nil`.